specs/validator/beacon_node_oapi.yaml

```diff
@@ -0,0 +1,613 @@
+            title: ProposerSlashings
+            type: object
+            description: "The [`ProposerSlashing`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#proposerslashing) object from the Eth2.0 spec."
+            properties:
+              proposer_index:
+                type: integer
+                format: uint64
+                description: "The index of the proposer to be slashed."
+              header_1:
+                $ref: '#/components/schemas/BeaconBlockHeader'
+              header_2:
+                $ref: '#/components/schemas/BeaconBlockHeader'
+        attester_slashings:
+          type: array
+          items:
+            title: AttesterSlashings
+            type: object
+            description: "The [`AttesterSlashing`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#attesterslashing) object from the Eth2.0 spec."
+            properties:
+              attestation_1:
+                $ref: '#/components/schemas/IndexedAttestation'
+              attestation_2:
+                $ref: '#/components/schemas/IndexedAttestation'
+        attestations:
+          type: array
+          items:
+            title: Attestation
+            type: object
+            description: "The [`Attestation`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#attestation) object from the Eth2.0 spec."
+            properties:
+              aggregation_bitfield:
+                type: string
+                format: byte
+                pattern: "^0x[a-fA-F0-9]+$"
+                description: "Attester aggregation bitfield."
+              custody_bitfield:
+                type: string
+                format: byte
+                pattern: "^0x[a-fA-F0-9]+$"
+                description: "Custody bitfield."
+              signature:
+                type: string
+                format: byte
+                pattern: "^0x[a-fA-F0-9]{192}$"
+                description: "BLS aggregate signature."
+              data:
+                $ref: '#/components/schemas/AttestationData'
+        deposits:
+          type: array
+          items:
+            title: Deposit
+            type: object
+            description: "The [`Deposit`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#deposit) object from the Eth2.0 spec."
+            properties:
+              proof:
+                type: array
+                description: "Branch in the deposit tree."
+                items:
+                  oneOf:
+                    - type: string
+                      format: byte
+                      pattern: "^0x[a-fA-F0-9]{64}$"
+                    - type: integer
+                      format: uint64
+                      enum: [32]
+                      example: 32
+                      description: "The DEPOSIT_CONTRACT_TREE_DEPTH value."
+                minItems: 2
+                maxItems: 2
```

Comment:
This is the only way I could figure out how to specify the [`Deposit`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#deposit) object, since it defines a two element array, where each element is a different type.

The definition I have provided is as restrictive as I could be while still enabling the spec. It says that it must be an array of length two, and items must either be bytes or the actual integer `32`.
Technically this means you could provide invalid things here, such as `[32, 32]` or `[bytes, bytes]`, but I wasn't sure how to be more specific in this case using OpenAPI 3. Certainly open to suggestion!

Reply:
> the Deposit object, since it defines a two element array

This is a [SSZ Vector](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/simple-serialize.md#composite-types) notation,
The notation of `'proof': ['bytes32', DEPOSIT_CONTRACT_TREE_DEPTH],` means SSZ vector of fixed size DEPOSIT_CONTRACT_TREE_DEPTH`

Reply:
> the Deposit object, since it defines a two element array

`'proof': ['bytes32', DEPOSIT_CONTRACT_TREE_DEPTH]` is a [SSZ Vector](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/simple-serialize.md#composite-types) notation, it means fixed-length (`DEPOSIT_CONTRACT_TREE_DEPTH`) array of `bytes32` values.

Now I see how confusing this notation is. 😬

Reply:
Ahh right! I thought it seemed strange. Thank you for clearing that up, I will push a fix for that.